dvc/dependency/repo.py

```diff
@@ -73,9 +75,18 @@ def fetch(self):
         return out
 
     def download(self, to):
-        out = self.fetch()
-        to.info = copy.copy(out.info)
-        to.checkout()
+        try:
+            out = self.fetch()
+            to.info = copy.copy(out.info)
+            to.checkout()
+        except NoOutputInExternalRepoError as e:
+            try:
+                with self._make_repo(
```

Comment:
Not sure what's an idiomatic way to avoid this duplication from fetch(), so I left it like this and I'm open to suggestions.